lib/url.ts

```diff
@@ -1,8 +1,56 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License. See License.txt in the project root for license information.
 
-export interface URLQuery {
-  [queryParameterName: string]: string;
+export class URLQuery {
+  private readonly _rawQuery: { [queryParameterName: string]: string } = {};
+
+  public any(): boolean {
+    return Object.keys(this._rawQuery).length > 0;
+  }
+
+  public set(parameterName: string, parameterValue: string | undefined): void {
+    if (parameterName) {
+      if (parameterValue) {
```

Comment:
what happens if the parameterValue is an empty string ? Empty string are falsy values.

Reply:
Yup. In that case I'll remove the parameter from the URLQuery.

Reply:
We talked about this, and since an empty string is a valid parameter value, I'll update this logic so that it checks if the parameter value is undefined/null instead.